solutions/usaco-920.mdx

```diff
@@ -2,13 +2,102 @@
 id: usaco-920
 source: USACO Silver 2019 February
 title: The Great Revegetation
-author: Sofia Yang
+author: Sofia Yang, Ryan Chou
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_revegetate_silver_feb19.html)
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_revegetate_silver_feb19.html)
 
+Assume that we have a valid solution composed of $K$ connected components in our graph.
+
+Now, let's look at one of these, which could look something like this:
+
+![possible connected component](<usaco-920/connected_component.png>)
+
+However, for all $K$ connected component, the grass types can be "inverted":
+
+![inverted connected component](<usaco-920/inverted_connected_component.png>)
+
+So, for each $K$ connected components, there are $2$ possible ways to arrange them, or $2^k$ ways to arrange the entire graph. Since in binary, $2^k$ is $1$ followed by $k$ zeros, we can simply print this instead of manually converting it.
```

Comment:
how about
"for each connected component in our graph..."
"thus, if there's k connected components, there's 2^k ways to arrange"